contracts/test-hs/test/Test/Swaps/Whitelisted.hs

```diff
@@ -0,0 +1,109 @@
+-- | Tests on the swaps contract.
+module Test.Swaps.Whitelisted
+  ( test_Whitelisted
+  ) where
+
+import Prelude hiding (swap)
+
+import Test.Tasty (TestTree, testGroup)
+
+import Morley.Nettest
+import Morley.Nettest.Tasty (nettestScenarioCaps)
+
+import Lorentz.Contracts.NonPausableSimpleAdmin (errNotAdmin)
+import Lorentz.Contracts.Swaps
+import Lorentz.Contracts.WhitelistedSwaps
+import Lorentz.Value
+import Test.Swaps.Util
+import Test.Util
+
+test_Whitelisted :: TestTree
+test_Whitelisted = testGroup "Whitelisted swap functionality"
+  [ whitelistUpdateAuthorization
+  , whitelistChecks
+  ]
```

Comment:
No need for the `test_Whitelisted` group. We can just prepend `test_` to the other tests.

Reply:
Initially, I went the same way you described for the tests on basic functionality but then realized that with separate `test_...` methods reusing the test suites gets complicated. Since whitelisting is, in theory, also a reusable component, I thought it may be better to go with grouping from the beginning (to make the future diff very small).

Reply:
Though on the other hand, it's really worth doing at least for demonstration purposes.

Reply:
Ah that makes sense, I'm ok either way in that case :)